### PR TITLE
Mobileapps 832

### PR DESCRIPTION
**Pull request by aman-alfresco**
1. Mimic app config json in app bundle.
2. Fetch app config from server and save in internal storage.
3. fetch app config on every app launch and if 24 hours completed.
4. Added time test cases

### CODE REVIEW

**Review by ankitgoyalgloballogic — CHANGES_REQUESTED** [review]
Peer review comments:
1. Function names should be generic. For example: you have created 2 functions like previous_time_passed_24_hours and previous_time_not_passed_24_hours. It should not contains 24_hours in the function name as this time can be changed.

2. Function isTimeToFetchConfig contains 24 hours check hardcoded. The time 24 hours should be fetched from some local file which can be changed at any given interval of time.

File: data/src/main/kotlin/com/alfresco/content/data/AppConfigUtils.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import java.lang.StringBuilder
+
+const val APP_CONFIG_JSON = "app.config.json"
+
+/**
+ * @property timeStamp
+ * This method check if the given timeStamp has passed the 24 hours or not
+ */
+fun isTimeToFetchConfig(timeStamp: Long): Boolean {
+    val hours24 = (1 * 24 * 60 * 60 * 1000)
```
Comment (gl-lovekesh) [inline data/src/main/kotlin/com/alfresco/content/data/AppConfigUtils.kt:22]: hours24 > hoursInMiliseconds
24 declare as constant.

**Review by gl-lovekesh — CHANGES_REQUESTED** [review]

File: data/src/main/kotlin/com/alfresco/content/data/AppConfigUtils.kt (hunk 1 above)
Comment (aman-alfresco) [reply data/src/main/kotlin/com/alfresco/content/data/AppConfigUtils.kt:22]: Updated it.

**Comment by aman-alfresco** [comment]
> Peer review comments:
> 
> 1. Function names should be generic. For example: you have created 2 functions like previous_time_passed_24_hours and previous_time_not_passed_24_hours. It should not contains 24_hours in the function name as this time can be changed.
> 2. Function isTimeToFetchConfig contains 24 hours check hardcoded. The time 24 hours should be fetched from some local file which can be changed at any given interval of time.

I've updated the requested changes

**Review by gl-lovekesh — APPROVED** [review]

**Pull request merged**